### ISSUE TITLE
No notifications on messages and calls for Android and iOS

### ISSUE THREAD

**Issue by ibahnasy** [signalapp/Signal-Android#13518]
- [ x] I have searched open and closed issues for duplicates
- [ x] I am submitting a bug report for existing functionality that does not work as intended
- [ x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Lots of times there is no notifications on new messages or calls on Android and iOS; specially Android.
When you check your phone, you could see a missed call notification.
This happens randomly, but certainly if Signal has not been used for sometime, even for few hours.
Background activity is allowed and battery optimization is disabled.

This report could be relevant: #9533

### Device info

**Device:** Samsung Galaxy A52 with /e/OS ROM
**Android version:** 13
**Signal version:** 7.3.1

**Comment by alex-signal** [comment]
Hi! Could you please include a debug log? If you don't wish to share one here, you can email a link to one via the app or directly to support at signal dot org. Please be sure to also mention this issue number.

**Comment by ibahnasy** [comment]
This is the log from the recipient's phone: 
[Receiver-signal-log-1713199986762.zip](https://github.com/signalapp/Signal-Android/files/14982176/Receiver-signal-log-1713199986762.zip)

This is the log from the caller's phone:
[Caller-signal-log-1713200072238.zip](https://github.com/signalapp/Signal-Android/files/14982177/Caller-signal-log-1713200072238.zip)

**Comment by misieck** [comment]
I have similar issue. After a period of inactivity, signal stops notifiyng about messages. Messages and notifications pop up only after manually starting signal.
No battery restrictions, data saver off.
OnePlus 5t with Lineage 20 (android 13).
20-20240205-microG-dumpling. Works on identical phone with original firmware.

**Comment by ibahnasy** [comment]
This is issue it not limited to messages only but calls too.

@alex-signal is there anything interesting in the logs?

**Comment by Brog33** [comment]
I have experienced this bug for ages in Android. I get all the messages but no notification sound/vibration. All settings seem to be correct and I'm not using battery optimising on the app. I really hope to be able to resolve this cos I've missed so many important msgs because of this bug.

**Comment by alex-signal** [comment]
@Brog33 Hi! Could you please upload a debug log? We've changed a number of things with how notifications work here but unfortunately sometimes device version / manufacturer prevent things from happening properly.

**Comment by Brog33** [comment]
> @Brog33 Hi! Could you please upload a debug log? We've changed a number of things with how notifications work here but unfortunately sometimes device version / manufacturer prevent things from happening properly.

Here you go. Hope you can figure it out. Just did a quick peek and saw this.

[7.11.4] [37656] 2024-08-02 10:43:53.601 GMT+02:00 I NotificationFactory: Security exception when posting notification, clearing ringtone
[7.11.4] [37656] 2024-08-02 10:43:53.603 GMT+02:00 I DefaultMessageNotifier: threads: 1 messages: 3
[7.11.4] [37656] 2024-08-02 10:43:53.604 GMT+02:00 E DefaultMessageNotifier: Notifications should be showing but are not for 1 threads


[signal-log-1722588305588.zip](https://github.com/user-attachments/files/16468240/signal-log-1722588305588.zip)

**Comment by Brog33** [comment]
Did you figure it out? Let me know if I can assist with more logs or testing.

**Comment by Brog33** [comment]
I got an in-app popup with instructions to disable battery saver etc on OnePlus. But having done all of those steps, it's still not showing any notification. None. Zilch.

**Comment by greyson-signal** [comment]
I dug through your logs you submitted on Aug 2nd. The lines you called out are indeed the most suspicious. We do occasionally have trouble fetching messages in the background, but not generally. 

Are there any device-specific settings around notifications that you can think of? Does anything in this guide help?
https://dontkillmyapp.com/oneplus

**Comment by Brog33** [comment]
Thanks. There was one setting in the guide that I hadn't turned off: Sleep Standby Optimization. I've disabled it and restarted the phone. Still no notifications. I don't know what else I can do.

**Comment by televators** [comment]
I just realized I haven't been getting notifications. I thought my group chat with my bff's was oddly quiet and it turned out I had missed texts going back two days.

I'm on Signal v7.17.6, Android 14, Samsung Galaxy S22. I do use Battery Saver about once a day in the evening but I have Signal set to be ignore for auto sleeping and whatnot. All notification settings normal, all permissions normal, and, like I said, Signal has an exception in Battery Saver settings to not get suspended.

As annoying as this issue is, I think it's the first bug I've encountered with Signal over three different phones in the past few years. So, pretty good track record for me.

**Comment by greyson-signal** [comment]
@televators can you share a link to a log? Settings > Help > Debuglog

**Comment by Brog33** [comment]
@greyson-signal so I have checked and re-checked every possible device-specific settings to prevent battery optimization and suppression of notifications. Unfortunately it doesn't make any difference to the missing notifications from Signal.

**Comment by greyson-signal** [comment]
@Brog33 That is a bug that should be fixed in 7.20.x

**Comment by Brog33** [comment]
I give up. I'm regrettably getting a new mobile, and it's only because of the lack of notifications in Signal.

**Comment by misieck** [comment]
@Brog33 What phone and os do you have exactly? I suspect this might be a bug in lineage. Doesnt exist on stock phone.

**Comment by Brog33** [comment]
@misieck OnePlus 7T (HD1903). Android 12 (OxygenOS 12.1)

**Comment by misieck** [comment]
Mine is 5T on lineage 20, so android 13. 
Still, it probably it is OS related. Friend has 5T with Oxygen 10 and no problem. 
If so this shouldnt be only a Signal problem. Except Signal is the only app I expect instant messaging from.
Do other apps work correctly? (i dont think sms counts - i suspect it takes a different path).

**Comment by Brog33** [comment]
@misieck Yeah, before the latest OxygenOS update (been a long while ago now) the notifications worked. I don't have any problems with notifications in other apps.

**Comment by wertzuz** [comment]
might be related to #8692

**Comment by yeah1227** [comment]
I had the same thing on ColorOS 16. Messages arrive, but no notification sound (make sure you have **auto launch** ON and **battery usage** to "Allow Background activity"). Than I had to go to **manage notifications** and setup a new ring tone for each of them (messages, additinal message, etc..) - one that isn't the default. I also went to premissions and removed "auto revoke premissions". Now I have no idea which one did the trick as I do not have time atm to test each setting and see what made it work, but it works now. Hope this helps someone.

**Comment by ar-siddiqui** [comment]
I have asked many of my friends and family to switch to Signal for messaging, they all complain about Signal not notifying them of messages, and can only see my message if they open Signal App.

They have other messaging apps that provide instant notification, so it doesn't seem like a setting issue. 

Some of those people are in countries where censorship circumvention is on.

**Comment by ibahnasy** [comment]
Same issue happening on Google Pixel and Xiaomi Redmi note 11.

**Comment by greyson-signal** [comment]
@ar-siddiqui 

> Some of those people are in countries where censorship circumvention is on.

In these countries, it is quite possible that FCM itself doesn't work. It's hard for us to be reliable with notifications in these regions.

For people in non-censored regions, I recommend reading our [support documentation](https://support.signal.org/hc/en-us/articles/360007318711-Troubleshooting-Notifications#android_notifications_troubleshooting) or checking out https://dontkillmyapp.com/

> They have other messaging apps that provide instant notification, so it doesn't seem like a setting issue.

Unfortunately it is an open secret that manufacturers will whitelist popular messaging apps from their battery optimizations, while punishing apps that are not on that whitelist by not allowing them to run in the background.

**Comment by ibahnasy** [comment]
I'm using /e/OS https://e.foundation
And notifications from all other apps like WhatsApp, Telegram etc are working fine.
Also Signal is the only app that doesn't incoming call notification in the smart watch!

**Comment by greyson-signal** [comment]
@ibahnasy Please verify that you've disabled battery optimizations for Signal. Also, please share a [log](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports), along with the timestamp of a message that was delayed (long press > Info > long press sent time, will copy timestamp to clipboard).

Regarding call notifications, we'll look into it 👍

**Comment by misieck** [comment]
Can it be that the battery setting somehow resets? I could have sworn i set it in one of my peoples phones only for them to stop receiving notifications and yes, the battery settings were wrong.. again.
As a positive note its been working on my phone for a long time now.
What is the technical reason we need to do this? Why the other apps don't need to fxxk around in settings for them to work?

**Comment by greyson-signal** [comment]
@misieck 

> What is the technical reason we need to do this? Why the other apps don't need to fxxk around in settings for them to work?

Depends on the app and the OEM. Some OEMs want to have the "best battery life", so they'll reject all background activity from all apps by default except for the most popular ones. If you're not on that list, tough luck.

For other OEMs, they may not like that Signal doesn't include the actual notification content in the FCM payload, which requires us to make a network request in order to retrieve messages, decrypt them, and render the notification. They'll restrict our network usage in background, preventing us from doing that.

But in general it's somewhere in the middle. For whatever reason, an OEM will occasionally delay our notifications due to some battery optimization technique that sometimes delays FCM delivery, and sometimes doesn't (perhaps based on whether your phone is idle, things like that). And similarly, they'll whitelist specific popular apps from these restrictions.

But in most circumstances, disabling battery optimizations for our app tends to fix things.
